### PR TITLE
Car 3298 refactoring

### PR DESCRIPTION
**Pull request by JudeNiroshan**
## Description

Following points have been refactored:

- the docs say that it retries 3 times [here](https://github.com/commercetools/commercetools-payone-integration/blob/2ea89c810bb40b5b6d2231aa0ae325b85dd28b92/service/src/main/java/com/commercetools/util/HttpRequestUtil.java#L80) but according to the code, it is 5. Adhere to the default values which are more similar to the HTTP client that is being used for CTP communication. Exceptions on the values may be acceptable as this is more critical on the business workflow. ✅
- the logged error does not say that is has been retried - would be useful so that one does not need to look up the source code every time ✅
- why retry and whole request creation is in a separate class HttpRequestUtil and not in PayonePostServiceImpl? Does the integration use this util for anything else than talking to Payone? Consider renaming the HttpRequestUtil class as this is only being used for payone HTTP client utilities. (Renamed to `PayoneHttpClientConfigurationUtil` ✅)
- Does it seem that [connectionTimeout](https://github.com/commercetools/commercetools-payone-integration/blob/2ea89c810bb40b5b6d2231aa0ae325b85dd28b92/service/src/main/java/com/commercetools/pspadapter/payone/domain/payone/PayonePostServiceImpl.java#L32) is not used in this class? (removed ✅)

### CODE REVIEW

File: functionaltests/src/test/java/specs/BaseFixture.java [hunk 1]
```diff
@@ -59,7 +60,8 @@
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
-import static com.commercetools.util.HttpRequestUtil.*;
+import static com.commercetools.pspadapter.payone.domain.payone.PayonePostServiceImpl.executeGetRequest;
+import static com.commercetools.util.PayoneHttpClientConfigurationUtil.*;
```
Comment (ahalberkamp) [inline functionaltests/src/test/java/specs/BaseFixture.java:64]: avoid wildcards for imports

File: functionaltests/src/test/java/specs/paymentmethods/creditcard/ChargePreauthorizedFixture.java [hunk 2]
```diff
@@ -40,7 +40,8 @@
 import java.util.Locale;
 import java.util.concurrent.ExecutionException;
 
-import static com.commercetools.util.HttpRequestUtil.executeGetRequest;
+import static com.commercetools.pspadapter.payone.domain.payone.PayonePostServiceImpl.executeGetRequest;
+
```
Comment (ahalberkamp) [inline functionaltests/src/test/java/specs/paymentmethods/creditcard/ChargePreauthorizedFixture.java:44]:
```suggestion
```

File: service/src/main/java/com/commercetools/pspadapter/payone/domain/payone/PayonePostServiceImpl.java [hunk 3]
```diff
@@ -17,20 +35,57 @@
...
+ * (see {@link DefaultHttpRequestRetryHandler#isRequestSentRetryEnabled()}
+ * and {@link PayoneHttpClientConfigurationUtil#httpRequestRetryHandler})</li>
+ * <li>connections pool is 200 connections, up to 20 per route (see {@link PayoneHttpClientConfigurationUtil#CONNECTION_MAX_TOTAL}
+ * and {@link PayoneHttpClientConfigurationUtil#CONNECTION_MAX_PER_ROUTE}). These values are "inherited" from
+ * <a href="https://github.com/Kong/unirest-java/blob/3b461599ad021d0a3f14213c0dbb85bab7244f66/src/main/java/com/mashape/unirest/http/options/Options.java#L23-L24">Unirest</a>
+ * library. It could be changed in the future if we face problems (for example, decrease if we have OutOfMemory
+ * or increase if out of connections from the pool.</li>
+ * </ul>
+ * <p>
+ * This util is intended to replace <i>Unirest</i> and <i>fluent-hc</i> dependencies, which don't propose any flexible
```
Comment (ahalberkamp) [inline service/src/main/java/com/commercetools/pspadapter/payone/domain/payone/PayonePostServiceImpl.java:58]:
```suggestion
 * This service is intended to replace <i>Unirest</i> and <i>fluent-hc</i> dependencies, which don't propose any flexible
```

File: service/src/main/java/com/commercetools/util/PayoneHttpClientConfigurationUtil.java [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+import java.util.Arrays;
+import java.util.Objects;
+
+import static com.commercetools.pspadapter.payone.domain.payone.PayonePostServiceImpl.executeGetRequest;
+import static java.lang.String.format;
+
+/**
+ * Util to make simple retryable HTTP GET/POST requests for Payone Http Client.
+ */
+public final class PayoneHttpClientConfigurationUtil {
```
Comment (ahalberkamp) [inline service/src/main/java/com/commercetools/util/PayoneHttpClientConfigurationUtil.java:33]: I would prefer this name
```suggestion
public final class PayoneHttpClientUtil {
```

File: service/src/test/java/com/commercetools/util/PayoneHttpClientConfigurationUtilParallelTest.java [hunk 5]
```diff
@@ -12,22 +12,24 @@
 import java.util.concurrent.atomic.AtomicInteger;
 import java.util.function.Supplier;
 
-import static com.commercetools.util.HttpRequestUtil.*;
-import static com.commercetools.util.HttpRequestUtilTest.*;
+import static com.commercetools.pspadapter.payone.domain.payone.PayonePostServiceImpl.executeGetRequest;
+import static com.commercetools.pspadapter.payone.domain.payone.PayonePostServiceImpl.executePostRequest;
+import static com.commercetools.util.PayoneHttpClientConfigurationUtil.*;
+import static com.commercetools.util.PayoneHttpClientConfigurationUtilTest.*;
```
Comment (ahalberkamp) [inline service/src/test/java/com/commercetools/util/PayoneHttpClientConfigurationUtilParallelTest.java:18]: avoid wildcards

File: service/src/test/java/com/commercetools/util/PayoneHttpClientConfigurationUtilTest.java [hunk 6]
```diff
@@ -10,14 +10,17 @@
 
 import java.util.Map;
 
-import static com.commercetools.util.HttpRequestUtil.*;
+import static com.commercetools.pspadapter.payone.domain.payone.PayonePostServiceImpl.executeGetRequest;
+import static com.commercetools.pspadapter.payone.domain.payone.PayonePostServiceImpl.executePostRequest;
+import static com.commercetools.pspadapter.payone.domain.payone.PayonePostServiceImpl.executePostRequestToString;
+import static com.commercetools.util.PayoneHttpClientConfigurationUtil.*;
```
Comment (ahalberkamp) [inline service/src/test/java/com/commercetools/util/PayoneHttpClientConfigurationUtilTest.java:16]: wildcard...

File: service/src/test/java/com/commercetools/util/PayoneHttpClientConfigurationUtilTimeoutTest.java [hunk 7]
```diff
@@ -17,13 +17,15 @@
 import java.util.List;
 import java.util.Random;
 
-import static com.commercetools.util.HttpRequestUtil.*;
+import static com.commercetools.pspadapter.payone.domain.payone.PayonePostServiceImpl.executeGetRequest;
+import static com.commercetools.pspadapter.payone.domain.payone.PayonePostServiceImpl.executePostRequest;
+import static com.commercetools.util.PayoneHttpClientConfigurationUtil.*;
```
Comment (ahalberkamp) [inline service/src/test/java/com/commercetools/util/PayoneHttpClientConfigurationUtilTimeoutTest.java:22]: wildcard

**Review by ahalberkamp — CHANGES_REQUESTED** [review]

File: functionaltests/src/test/java/specs/BaseFixture.java (hunk 1 above)
Comment (JudeNiroshan) [reply functionaltests/src/test/java/specs/BaseFixture.java]: d366c04

File: functionaltests/src/test/java/specs/paymentmethods/creditcard/ChargePreauthorizedFixture.java (hunk 2 above)
Comment (JudeNiroshan) [reply functionaltests/src/test/java/specs/paymentmethods/creditcard/ChargePreauthorizedFixture.java:44]: c41fdca

File: service/src/main/java/com/commercetools/pspadapter/payone/domain/payone/PayonePostServiceImpl.java (hunk 3 above)
Comment (JudeNiroshan) [reply service/src/main/java/com/commercetools/pspadapter/payone/domain/payone/PayonePostServiceImpl.java]: d17c602

File: service/src/main/java/com/commercetools/util/PayoneHttpClientConfigurationUtil.java (hunk 4 above)
Comment (JudeNiroshan) [reply service/src/main/java/com/commercetools/util/PayoneHttpClientConfigurationUtil.java]: 1a87206

File: service/src/test/java/com/commercetools/util/PayoneHttpClientConfigurationUtilParallelTest.java (hunk 5 above)
Comment (JudeNiroshan) [reply service/src/test/java/com/commercetools/util/PayoneHttpClientConfigurationUtilParallelTest.java]: f4058fa

File: service/src/test/java/com/commercetools/util/PayoneHttpClientConfigurationUtilTest.java (hunk 6 above)
Comment (JudeNiroshan) [reply service/src/test/java/com/commercetools/util/PayoneHttpClientConfigurationUtilTest.java]: 292b534

File: service/src/test/java/com/commercetools/util/PayoneHttpClientConfigurationUtilTimeoutTest.java (hunk 7 above)
Comment (JudeNiroshan) [reply service/src/test/java/com/commercetools/util/PayoneHttpClientConfigurationUtilTimeoutTest.java]: ecfc5bd

**Review by ahalberkamp — APPROVED** [review]

**Pull request merged**